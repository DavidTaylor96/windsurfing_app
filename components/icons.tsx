import { Feather } from '@expo/vector-icons';
import React, { FC } from 'react';

export type IconType = keyof typeof Feather.glyphMap;
export interface IIconProps {
  icon: IconType;
  color?: string;
  size?: number;
}

export const Icon: FC<IIconProps> = (props) => {
  return (
    <Feather
      name={props.icon}
      size={props.size ? props.size : 24}
      color={props.color ? props.color : 'black'}
    />
  );
};
