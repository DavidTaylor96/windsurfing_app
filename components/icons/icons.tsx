import { Feather } from '@expo/vector-icons';
import React, { FC } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

export type IconType = keyof typeof Feather.glyphMap;
export interface IIconProps {
  icon: IconType;
  color?: string;
  size?: number;
  style?: StyleProp<ViewStyle>;
}

export const Icon: FC<IIconProps> = (props) => {
  return (
    <Feather
      name={props.icon}
      size={props.size ? props.size : 24}
      color={props.color ? props.color : 'black'}
      style={props.style}
    />
  );
};
