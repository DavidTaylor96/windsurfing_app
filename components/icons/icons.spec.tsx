import * as React from 'react';
import renderer from 'react-test-renderer';
import { Icon } from './icons';

it(`Icons`, () => {

  const icon = renderer.create(
    <Icon icon={'activity'} color={'blue'} size={15} />
  );

  expect(icon).toMatchSnapshot();
});
