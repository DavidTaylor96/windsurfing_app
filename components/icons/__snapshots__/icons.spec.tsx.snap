// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons 1`] = `<Text />`;
